p0f/app.py

```diff
@@ -27,30 +29,40 @@ def send_rabbit_msg(msg, channel, exchange='', routing_key='task_queue'):
     return
 
 def get_version():
-    version = ''
-    with open('VERSION', 'r') as f:
-        for line in f:
-            version = line.strip()
-    return version
+    with open(os.path.join(os.path.dirname(os.path.realpath(__file__)), 'VERSION'), 'r') as f:
+        return f.read().strip()
 
-def run_p0f(path):
-    os.system('/usr/bin/p0f -r ' + path + ' -o /tmp/p0f_output.txt > /dev/null')
-    return
+def run_proc(args, output=subprocess.DEVNULL):
+    proc = subprocess.Popen(args, stdout=output)
+    return proc.communicate()
 
-def run_tshark(path):
-    os.system('/usr/bin/tshark -r ' + path + ' -T fields -e ip.src -e eth.src | sort | uniq > /tmp/tshark_output.txt')
-    os.system('/usr/bin/tshark -r ' + path + ' -T fields -e ip.dst -e eth.dst | sort | uniq >> /tmp/tshark_output.txt')
-    return
+def run_p0f(path, p0f_output, p0f='/usr/bin/p0f'):
+    args = [p0f, '-r', path, '-o', p0f_output]
+    return run_proc(args)
 
-def parse_output():
+def run_tshark(path, tshark_output, tshark='/usr/bin/tshark'):
+    exit_status = []
+    with tempfile.TemporaryDirectory() as tempdir:
+        tshark_tmp = os.path.join(tempdir, 'fields.txt')
+        with open(tshark_tmp, 'w') as output:
+            args = [tshark, '-r', path, '-T', 'fields', '-e', 'eth.src', '-e', 'ip.src']
```

Comment:
why is `| sort | uniq` no longer needed?

Reply:
Because I use python set() to read the raw output and do the unique operation in python.
I won't even need to do that once I move to pyshark.

Reply:
ah, missed that. cool